docs/secure-kafka/_index.md

```diff
@@ -0,0 +1,552 @@
++           password="xxxxxxx"
++       ;',
+
+    'properties.group.id' = 'sales-record-group',
+    'value.format' = 'avro-confluent',
+    'value.avro-confluent.url' = 'http://apicurio-registry-service.flink.svc:8080/apis/ccompat/v6',
+    'scan.startup.mode' = 'latest-offset'
+);
+```
+
+### OAuth 2.0
+
+> Note:
+> - This example uses the Keycloak config and realm from the
+> [`strimzi-kafka-operator` Keycloak example](https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/security/keycloak-authorization/kafka-authz-realm.json).
+>   - Username: `admin`
+>   - Password: `admin`
+
+```shell
+# Note: For OAuth 2.0, Keycloak and a self-signed
+# TLS certificate for Keycloak are automatically created.
+SECURE_KAFKA=OAuth2 ./scripts/data-gen-setup.sh
+
+kubectl -n flink apply -f secure-kafka/OAuth2/standalone-etl-secure-deployment.yaml
+```
+
+The commands above apply the following:
+
+```diff
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    listeners:
+-     - name: plain
+-       port: 9092
+-       type: internal
+-       tls: false
+
++     - name: oauth2
++       port: 9094
++       type: internal
++       tls: true
++       authentication:
++         type: oauth       # Specify OAuth 2.0 authentication
+
++           # Specify OAuth 2.0 JWKS/JWT details
++         validIssuerUri: https://keycloak.flink.svc:8443/realms/kafka-authz
++         jwksEndpointUri: https://keycloak.flink.svc:8443/realms/kafka-authz/protocol/openid-connect/certs
++         userNameClaim: preferred_username
+
++           # Trust self-signed TLS certificate used by Keycloak HTTPS endpoint
++         tlsTrustedCertificates:
++           - secretName: keycloak-cert
++             certificate: tls.crt
+```
+
+```diff
+# This KafkaUser is only needed if using simple authorization
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaUser
+metadata:
+- name: my-user
++ name: service-account-kafka   # Create KafkaUser for user in the Keycloak realm
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+- authentication:               # Remove authentication
+-   type: tls
+```
+
+```diff
+CREATE TABLE SalesRecordTable (
+    invoice_id STRING,
+    user_id STRING,
+    product_id STRING,
+    quantity STRING,
+    unit_cost STRING,
+    `purchase_time` TIMESTAMP(3) METADATA FROM 'timestamp',
+    WATERMARK FOR purchase_time AS purchase_time - INTERVAL '1' SECOND
+) WITH ( 
+    'connector' = 'kafka',
+    'topic' = 'flink.sales.records',
+-   'properties.bootstrap.servers' = 'my-cluster-kafka-bootstrap.flink.svc:9092',
++   'properties.bootstrap.servers' = 'my-cluster-kafka-bootstrap.flink.svc:9094',
++   'properties.security.protocol' = 'SASL_SSL',
++   'properties.ssl.truststore.location' = '/opt/my-cluster-cluster-ca-cert/ca.crt',
++   'properties.ssl.truststore.type' = 'PEM',
+
++     -- Connect using OAUTHBEARER mechanism (OAuth 2.0 with Bearer token)
++   'properties.sasl.mechanism' = 'OAUTHBEARER',
```

Comment:
I don't think it is worth a whole other section on the `PLAIN` setup, but a link to the docs (in a intro line/paragraph) above would be good.